http/src/main.rs

```diff
@@ -185,17 +190,29 @@ fn main() {
 
 fn serve<Types: IpfsTypes>(
     ipfs: &Ipfs<Types>,
+    listening_addr: Multiaddr,
 ) -> (std::net::SocketAddr, impl std::future::Future<Output = ()>) {
     use tokio::stream::StreamExt;
     use warp::Filter;
+
     let (shutdown_tx, mut shutdown_rx) = tokio::sync::mpsc::channel::<()>(1);
 
     let routes = v0::routes(ipfs, shutdown_tx);
     let routes = routes.with(warp::log(env!("CARGO_PKG_NAME")));
 
     let ipfs = ipfs.clone();
 
-    warp::serve(routes).bind_with_graceful_shutdown(([127, 0, 0, 1], 0), async move {
+    let components = listening_addr.iter().collect::<Vec<_>>();
+
+    use std::net::SocketAddr;
```

Comment:
I wonder if this could be moved to the beginning of the function, like the rest of the `use`s?

Reply:
Alternative would be to not `use` but to have the full path on the single use.